crates/biome_js_analyze/src/semantic_analyzers/correctness/no_unused_variables.rs

```diff
@@ -377,3 +378,25 @@ impl Rule for NoUnusedVariables {
         }
     }
 }
+
+/// Returns `true` if `expr` is unused.
+fn is_unused_expression(expr: &JsSyntaxNode) -> SyntaxResult<bool> {
+    debug_assert!(AnyJsExpression::can_cast(expr.kind()));
+    let mut previous = expr.text_range();
```

Comment:
Could you help me to understand this function?
What do we want to check?
Is it possible to compare nodes instead of text range?

Reply:
This function determines if the expression `expr` is in a position that makes it unused.
This is the acse if:

- the expression is a  statement
- the expression is not at the end of a comma sequence

Some examples:

```js
(expr);
const x = (expr, 0);
```

And some other examples where `expr` is determined as used:

```js
const x = (expr);
const y = (0, expr);
```

> Is it possible to compare nodes instead of text ranges?

Yes. the comparison is to determine if the expression is not at the end of a comma sequence.

Reply:
I can remove the comma sequence check. It is a rare case.

Reply:
I don't know actually,
I think it's okey to have it.